### ISSUE TITLE
Cannot build from docker compose

### ISSUE THREAD

**Issue by rsandrini** [HarukaMa/aleo-explorer#22]
So I tried to build from the docker compose file, but cffi is failing in the building process

```
19.88 Building wheels for collected packages: minify-html, cffi
19.88   Building wheel for minify-html (pyproject.toml): started
86.33   Building wheel for minify-html (pyproject.toml): still running...
218.6   Building wheel for minify-html (pyproject.toml): still running...
295.5   Building wheel for minify-html (pyproject.toml): still running...
295.8   Building wheel for minify-html (pyproject.toml): finished with status 'done'
295.8   Created wheel for minify-html: filename=minify_html-0.11.5-cp311-cp311-linux_aarch64.whl size=745669 sha256=9c92da54afc409c45fe8ca3d8f37f75751aa8eeaec4e976001aa5ae9e68c76e4
295.8   Stored in directory: /root/.cache/pip/wheels/7a/9c/31/f30e452c46b8288b7a2bee2c52dde60828f710172e9600a9dc
295.8   Building wheel for cffi (pyproject.toml): started
296.7   Building wheel for cffi (pyproject.toml): finished with status 'error'
296.7   error: subprocess-exited-with-error
296.7
296.7   × Building wheel for cffi (pyproject.toml) did not run successfully.
296.7   │ exit code: 1
296.7   ╰─> [61 lines of output]
296.7       Package libffi was not found in the pkg-config search path.
296.7       Perhaps you should add the directory containing `libffi.pc'
296.7       to the PKG_CONFIG_PATH environment variable
296.7       Package 'libffi', required by 'virtual:world', not found
296.7       Package libffi was not found in the pkg-config search path.
296.7       Perhaps you should add the directory containing `libffi.pc'
296.7       to the PKG_CONFIG_PATH environment variable
296.7       Package 'libffi', required by 'virtual:world', not found
296.7       Package libffi was not found in the pkg-config search path.
296.7       Perhaps you should add the directory containing `libffi.pc'
296.7       to the PKG_CONFIG_PATH environment variable
296.7       Package 'libffi', required by 'virtual:world', not found
296.7       Package libffi was not found in the pkg-config search path.
296.7       Perhaps you should add the directory containing `libffi.pc'
296.7       to the PKG_CONFIG_PATH environment variable
296.7       Package 'libffi', required by 'virtual:world', not found
296.7       Package libffi was not found in the pkg-config search path.
296.7       Perhaps you should add the directory containing `libffi.pc'
296.7       to the PKG_CONFIG_PATH environment variable
296.7       Package 'libffi', required by 'virtual:world', not found
296.7       running bdist_wheel
296.7       running build
296.7       running build_py
296.7       creating build
296.7       creating build/lib.linux-aarch64-cpython-311
296.7       creating build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/pkgconfig.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/error.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/recompiler.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/vengine_gen.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/ffiplatform.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/commontypes.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/api.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/lock.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/_imp_emulation.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/backend_ctypes.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/vengine_cpy.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/setuptools_ext.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/cffi_opcode.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/__init__.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/_shimmed_dist_utils.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/verifier.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/model.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/cparser.py -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/_cffi_include.h -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/parse_c_type.h -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/_embedding.h -> build/lib.linux-aarch64-cpython-311/cffi
296.7       copying src/cffi/_cffi_errors.h -> build/lib.linux-aarch64-cpython-311/cffi
296.7       warning: build_py: byte-compiling is disabled, skipping.
296.7
296.7       running build_ext
296.7       building '_cffi_backend' extension
296.7       creating build/temp.linux-aarch64-cpython-311
296.7       creating build/temp.linux-aarch64-cpython-311/src
296.7       creating build/temp.linux-aarch64-cpython-311/src/c
296.7       gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DFFI_BUILDING=1 -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/usr/local/include/python3.11 -c src/c/_cffi_backend.c -o build/temp.linux-aarch64-cpython-311/src/c/_cffi_backend.o
296.7       src/c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
296.7          15 | #include <ffi.h>
296.7             |          ^~~~~~~
296.7       compilation terminated.
296.7       error: command '/usr/bin/gcc' failed with exit code 1
296.7       [end of output]
296.7
296.7   note: This error originates from a subprocess, and is likely not a problem with pip.
296.7   ERROR: Failed building wheel for cffi
296.7 Successfully built minify-html
296.7 Failed to build cffi
296.7 ERROR: Failed to build one or more wheels
------
failed to solve: process "/bin/sh -c set -eux;     pip wheel -w /dist/  -r /app/requirements.txt" did not complete successfully: exit code: 1
```

I'm building a Ubuntu machine from AWS
CPU ARMv8 (AWS Graviton2)

any suggestion on how to fix it?

**Comment by HarukaMa** [comment]
it did install libffi-dev so actually not sure right now why it couldnt find the header. havent really tried to run on arm devices.

**Comment by pa-long** [comment]
Fixed this by updating `apk add --no-cache` command in `alpine.Dockerfile`:
```Dockerfile

RUN set -eux; \
    apk add --no-cache \
    git \
    curl \
    bash \
    gcc \
    musl-dev \
    pkgconfig \
    openssl-dev \
    libc-dev \
    linux-headers \
    libffi-dev \
    ;
```

**Comment by HarukaMa** [comment]
Should be fixed by #31. Thanks @pa-long !
